UniConnectApp/app/src/main/java/com/example/uniconnect/service/UniversityService.kt

```diff
@@ -8,6 +8,11 @@ import kotlinx.coroutines.async
 import kotlinx.coroutines.withContext
 import retrofit2.awaitResponse
 
+interface IUniversityService {
+    suspend fun fetchUniversities(): List<University>?
+
+}
+
```

Comment:
Delete the IUniversityService and ddd this in the in the UniversityService. You can just code this here.

Reply:
Ehhhh, I'd keep them separate.